File: python3/missing_number.py
# Space : O(1)
# Time  : O(n)

class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        total_nums, total_idx = 0, 0

        for i in range(len(nums)):
            total_nums += nums[i]
            total_idx += i + 1

        return total_idx - total_nums

File: golang/missing_number.go
package main

// Space : O(1)
// Time  : O(n)

func missingNumber(nums []int) int {
	totalNums, totalRange := 0, 0
	n := len(nums)

	for i := 0; i < n; i++ {
		totalNums += nums[i]
		totalRange += i + 1
	}

	return totalRange - totalNums
}
